### PR TITLE
Fix double-freed global variable in easylogging lib

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: longjiquan <jiquan.long@zilliz.com>
/kind bug
issue: #17639

### CODE REVIEW

**Comment by cydrain** [comment]
I think it's not necessary to build a separate "milvus_log.so", the 2 files under directory "log" can be moved to "common".
You can see #16761

**Comment by xiaofan-luan** [comment]
> I think it's not necessary to build a separate "milvus_log.so", the 2 files under directory "log" can be moved to "common". You can see #16761

You can not depend on two different Easylog, that's why we crash.
So  we decide to only rely on Easylog in knowhere

**Comment by longjiquan** [comment]
> I think it's not necessary to build a separate "milvus_log.so", the 2 files under directory "log" can be moved to "common". You can see #16761

I'll open another pr to merge these two libraries if neccessary, since this is not the root cause why our program exits with core dump.

File: internal/core/src/log/CMakeLists.txt [hunk 1]
```diff
@@ -12,10 +12,14 @@
 #-------------------------------------------------------------------------------
 set(LOG_FILES   ${MILVUS_ENGINE_SRC}/log/Log.cpp
                 ${MILVUS_ENGINE_SRC}/log/Log.h
-                ${MILVUS_THIRDPARTY_SRC}/easyloggingpp/easylogging++.cc
-                ${MILVUS_THIRDPARTY_SRC}/easyloggingpp/easylogging++.h
+                #${MILVUS_THIRDPARTY_SRC}/easyloggingpp/easylogging++.cc
+                #${MILVUS_THIRDPARTY_SRC}/easyloggingpp/easylogging++.h
                 )
 
-add_library(milvus_log STATIC ${LOG_FILES})
```
[the remark this answers is not in the document]
Comment (longjiquan) [reply internal/core/src/log/CMakeLists.txt:19]: OK.

File: internal/core/src/log/CMakeLists.txt [hunk 2]
```diff
@@ -12,10 +12,14 @@
...
+add_library(milvus_log SHARED ${LOG_FILES})
 set_target_properties(milvus_log PROPERTIES RULE_LAUNCH_COMPILE "")
 set_target_properties(milvus_log PROPERTIES RULE_LAUNCH_LINK "")
+# milvus_log depends on symbols of easylogging in knowhere
+# TODO: remove this module and directly wrap log module in knowhere?
+target_link_libraries(milvus_log knowhere)
```
[the remark this answers is not in the document]
Comment (longjiquan) [reply internal/core/src/log/CMakeLists.txt:24]: OK.

[the remark this answers is not in the document]
Comment (longjiquan) [reply internal/core/src/log/CMakeLists.txt]: fixed.

File: internal/core/src/log/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (longjiquan) [reply internal/core/src/log/CMakeLists.txt:19]: fixed.

**Comment by Presburger** [comment]
LGTM

**Pull request merged**